### PR TITLE
Add support to use temporary security credentials

### PR DESCRIPTION
**Pull request by Biruntha**
## Purpose
Fix for issue 
https://github.com/wso2-ballerina/module-amazons3/issues/14

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by Biruntha** [comment]
Doc comments were already applied with the link of the ec2 PR
https://github.com/wso2-ballerina/module-amazonec2/pull/17

File: amazons3/Module.md [hunk 1]
```diff
@@ -31,23 +31,67 @@ First, import the `wso2/amazons3` module into the Ballerina project.
 import wso2/amazons3;
 ```
     
-The Amazon S3 connector can be instantiated using the accessKeyId, secretAccessKey, region, 
+The Amazon S3 connector can be instantiated using the accessKeyId, secretAccessKey, securityToken, and region
```
Comment (MaryamZi) [inline amazons3/Module.md]:
```suggestion
The Amazon S3 connector can be instantiated using the accessKeyId, secretAccessKey, securityToken, region
```

we have an "and" in the next line.

File: amazons3/Module.md [hunk 2]
```diff
@@ -31,23 +31,67 @@ First, import the `wso2/amazons3` module into the Ballerina project.
 import wso2/amazons3;
 ```
     
-The Amazon S3 connector can be instantiated using the accessKeyId, secretAccessKey, region, 
+The Amazon S3 connector can be instantiated using the accessKeyId, secretAccessKey, securityToken, region
```
Comment (MaryamZi) [inline amazons3/Module.md]:
```suggestion
The Amazon S3 connector can be instantiated using the accessKeyId, secretAccessKey, securityToken, region,
```

File: amazons3/amazons3_utils.bal [hunk 3]
```diff
@@ -102,7 +105,18 @@ function generateSignature(http:Request request, string accessKeyId, string secr
     canonicalHeaders = canonicalHeaders + ":";
     canonicalHeaders = canonicalHeaders + request.getHeader(X_AMZ_DATE.toLower());
     canonicalHeaders = canonicalHeaders + "\n";
+
+    if (securityToken != "") {
+        canonicalHeaders = canonicalHeaders + X_AMZ_SECURITY_TOKEN.toLower();
+        canonicalHeaders = canonicalHeaders + ":";
+        canonicalHeaders = canonicalHeaders + request.getHeader(X_AMZ_SECURITY_TOKEN.toLower());
+        canonicalHeaders = canonicalHeaders + "\n";
```
Comment (MaryamZi) [inline amazons3/amazons3_utils.bal]: can't we do all these concatenations in one line?

File: amazons3/amazons3_endpoint.bal [hunk 4]
```diff
@@ -40,6 +44,10 @@ public type Client client object {
         string baseURL = HTTPS + amazonS3Config.amazonHost;
         self.accessKeyId = amazonS3Config.accessKeyId;
         self.secretAccessKey = amazonS3Config.secretAccessKey;
+        var token = amazonS3Config.securityToken;
+        if ((token is string) && token != "") {
```
Comment (MaryamZi) [inline amazons3/amazons3_endpoint.bal]: Do we need the is empty check? 

We should either just set whatever the user sets, or return an error for empty strings, IMO.

Comment (Biruntha) [reply amazons3/amazons3_endpoint.bal]: yes we need. bcz when we pass root credentials, we don't need to set securityToken in the header. In that case if we haven't set any value for securityToken , the empty value is set. and we can't return error for empty string as well.

**Pull request merged**